In [1]:
from bs4 import BeautifulSoup  
import requests
import pandas as pd

Q1.Write a python program to display all the header tags from‘en.wikipedia.org/wiki/Main_Page’.

In [2]:
wiki_page = requests.get("https://en.wikipedia.org/wiki/Main_Page")
wiki_page

<Response [200]>

In [3]:
soup = BeautifulSoup(wiki_page.text, 'html.parser')
wiki_header = soup.find_all(['h1', 'h2','h3'])
wiki_header

[<h1 class="firstHeading" id="firstHeading">Main Page</h1>,
 <h2 class="mp-h2" id="mp-tfa-h2"><span id="From_today.27s_featured_article"></span><span class="mw-headline" id="From_today's_featured_article">From today's featured article</span></h2>,
 <h2 class="mp-h2" id="mp-dyk-h2"><span class="mw-headline" id="Did_you_know_...">Did you know ...</span></h2>,
 <h2 class="mp-h2" id="mp-itn-h2"><span class="mw-headline" id="In_the_news">In the news</span></h2>,
 <h2 class="mp-h2" id="mp-otd-h2"><span class="mw-headline" id="On_this_day">On this day</span></h2>,
 <h2 class="mp-h2" id="mp-tfp-h2"><span id="Today.27s_featured_picture"></span><span class="mw-headline" id="Today's_featured_picture">Today's featured picture</span></h2>,
 <h2 class="mp-h2" id="mp-other"><span class="mw-headline" id="Other_areas_of_Wikipedia">Other areas of Wikipedia</span></h2>,
 <h2 class="mp-h2" id="mp-sister"><span id="Wikipedia.27s_sister_projects"></span><span class="mw-headline" id="Wikipedia's_sister_proje

In [4]:
wiki_header1 = []
for i in wiki_header:
    wiki_header1.append(i.text.replace('\n',''))
print(" Total headers =",len(wiki_header1))
wiki_header1

 Total headers = 22


['Main Page',
 "From today's featured article",
 'Did you know\xa0...',
 'In the news',
 'On this day',
 "Today's featured picture",
 'Other areas of Wikipedia',
 "Wikipedia's sister projects",
 'Wikipedia languages',
 'Navigation menu',
 'Personal tools',
 'Namespaces',
 'Variants',
 'Views',
 'More',
 'Search',
 'Navigation',
 'Contribute',
 'Tools',
 'Print/export',
 'In other projects',
 'Languages']

In [5]:
headers = pd.DataFrame({})
headers["header"]=wiki_header1

In [6]:
headers

header
0                       Main Page
1   From today's featured article
2                Did you know ...
3                     In the news
4                     On this day
5        Today's featured picture
6        Other areas of Wikipedia
7     Wikipedia's sister projects
8             Wikipedia languages
9                 Navigation menu
10                 Personal tools
11                     Namespaces
12                       Variants
13                          Views
14                           More
15                         Search
16                     Navigation
17                     Contribute
18                          Tools
19                   Print/export
20              In other projects
21                      Languages

Q2.Write a python program to display IMDB’s Top rated 100 movies’ data (i.e. Name, IMDB rating, Year of 
release).

In [7]:
page_imdb = requests.get("https://www.imdb.com/list/ls091520106/")
page_imdb

<Response [200]>

In [8]:
soup = BeautifulSoup(page_imdb.text, 'html.parser')

In [9]:
m_imbd =soup.find_all("h3",class_="lister-item-header")
m_imbd

[<h3 class="lister-item-header">
 <span class="lister-item-index unbold text-primary">1.</span>
 <a href="/title/tt0111161/">The Shawshank Redemption</a>
 <span class="lister-item-year text-muted unbold">(1994)</span>
 </h3>,
 <h3 class="lister-item-header">
 <span class="lister-item-index unbold text-primary">2.</span>
 <a href="/title/tt0068646/">The Godfather</a>
 <span class="lister-item-year text-muted unbold">(1972)</span>
 </h3>,
 <h3 class="lister-item-header">
 <span class="lister-item-index unbold text-primary">3.</span>
 <a href="/title/tt0071562/">The Godfather: Part II</a>
 <span class="lister-item-year text-muted unbold">(1974)</span>
 </h3>,
 <h3 class="lister-item-header">
 <span class="lister-item-index unbold text-primary">4.</span>
 <a href="/title/tt0468569/">The Dark Knight</a>
 <span class="lister-item-year text-muted unbold">(2008)</span>
 </h3>,
 <h3 class="lister-item-header">
 <span class="lister-item-index unbold text-primary">5.</span>
 <a href="/title/tt005

In [10]:
movies_name =[]
for i in  m_imbd:
    for j in i.find_all("a"):
         movies_name.append(j.text.replace("\n",""))
movies_name      

['The Shawshank Redemption',
 'The Godfather',
 'The Godfather: Part II',
 'The Dark Knight',
 '12 Angry Men',
 "Schindler's List",
 'The Lord of the Rings: The Return of the King',
 'Pulp Fiction',
 'Il buono, il brutto, il cattivo',
 'Fight Club',
 'Joker',
 'The Lord of the Rings: The Fellowship of the Ring',
 'Forrest Gump',
 'Inception',
 'Star Wars: Episode V - The Empire Strikes Back',
 'The Lord of the Rings: The Two Towers',
 'The Matrix',
 "One Flew Over the Cuckoo's Nest",
 'Goodfellas',
 'Shichinin no samurai',
 'Se7en',
 'Cidade de Deus',
 'La vita è bella',
 'The Silence of the Lambs',
 'Star Wars',
 "It's a Wonderful Life",
 'Saving Private Ryan',
 'Sen to Chihiro no kamikakushi',
 'The Green Mile',
 'Léon',
 'Seppuku',
 'Interstellar',
 'The Usual Suspects',
 'The Lion King',
 'American History X',
 'Back to the Future',
 'The Pianist',
 'Modern Times',
 'Terminator 2: Judgment Day',
 'The Intouchables',
 'Psycho',
 'Gladiator',
 'City Lights',
 'The Departed',
 'Whipla

In [11]:
year =[]
for i in  m_imbd:
    for j in i.find_all("span"):
         year.append(j.text.replace("\n",""))
year

['1.',
 '(1994)',
 '2.',
 '(1972)',
 '3.',
 '(1974)',
 '4.',
 '(2008)',
 '5.',
 '(1957)',
 '6.',
 '(1993)',
 '7.',
 '(2003)',
 '8.',
 '(1994)',
 '9.',
 '(1966)',
 '10.',
 '(1999)',
 '11.',
 '(2019)',
 '12.',
 '(2001)',
 '13.',
 '(1994)',
 '14.',
 '(2010)',
 '15.',
 '(1980)',
 '16.',
 '(2002)',
 '17.',
 '(1999)',
 '18.',
 '(1975)',
 '19.',
 '(1990)',
 '20.',
 '(1954)',
 '21.',
 '(1995)',
 '22.',
 '(2002)',
 '23.',
 '(1997)',
 '24.',
 '(1991)',
 '25.',
 '(1977)',
 '26.',
 '(1946)',
 '27.',
 '(1998)',
 '28.',
 '(2001)',
 '29.',
 '(1999)',
 '30.',
 '(1994)',
 '31.',
 '(1962)',
 '32.',
 '(2014)',
 '33.',
 '(1995)',
 '34.',
 '(1994)',
 '35.',
 '(1998)',
 '36.',
 '(1985)',
 '37.',
 '(2002)',
 '38.',
 '(1936)',
 '39.',
 '(1991)',
 '40.',
 '(2011)',
 '41.',
 '(1960)',
 '42.',
 '(2000)',
 '43.',
 '(1931)',
 '44.',
 '(2006)',
 '45.',
 '(2014)',
 '46.',
 '(1968)',
 '47.',
 '(2006)',
 '48.',
 '(2019)',
 '49.',
 '(1942)',
 '50.',
 '(1988)',
 '51.',
 '(1954)',
 '52.',
 '(1988)',
 '53.',
 '(1979)',
 '

In [12]:
release_year =[]
for i in range(1,len(year),2):
    release_year.append(year[i])
release_year    

['(1994)',
 '(1972)',
 '(1974)',
 '(2008)',
 '(1957)',
 '(1993)',
 '(2003)',
 '(1994)',
 '(1966)',
 '(1999)',
 '(2019)',
 '(2001)',
 '(1994)',
 '(2010)',
 '(1980)',
 '(2002)',
 '(1999)',
 '(1975)',
 '(1990)',
 '(1954)',
 '(1995)',
 '(2002)',
 '(1997)',
 '(1991)',
 '(1977)',
 '(1946)',
 '(1998)',
 '(2001)',
 '(1999)',
 '(1994)',
 '(1962)',
 '(2014)',
 '(1995)',
 '(1994)',
 '(1998)',
 '(1985)',
 '(2002)',
 '(1936)',
 '(1991)',
 '(2011)',
 '(1960)',
 '(2000)',
 '(1931)',
 '(2006)',
 '(2014)',
 '(1968)',
 '(2006)',
 '(2019)',
 '(1942)',
 '(1988)',
 '(1954)',
 '(1988)',
 '(1979)',
 '(1981)',
 '(2000)',
 '(1979)',
 '(1940)',
 '(2006)',
 '(2018)',
 '(2012)',
 '(2018)',
 '(1980)',
 '(1957)',
 '(2008)',
 '(1950)',
 '(1964)',
 '(1997)',
 '(2003)',
 '(1957)',
 '(2012)',
 '(1984)',
 '(2019)',
 '(1986)',
 '(1999)',
 '(I) (2017)',
 '(2016)',
 '(1995)',
 '(1981)',
 '(2009)',
 '(2007)',
 '(1983)',
 '(1995)',
 '(1992)',
 '(1984)',
 '(2016)',
 '(1997)',
 '(2009)',
 '(1931)',
 '(2000)',
 '(1968)',
 '(195

In [13]:
m_imbd1 =soup.find_all("div",class_="ipl-rating-widget")
m_imbd1

[<div class="ipl-rating-widget">
 <div class="ipl-rating-star small">
 <span class="ipl-rating-star__star">
 <svg class="ipl-icon ipl-star-icon" fill="#000000" height="24" viewbox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg">
 <path d="M0 0h24v24H0z" fill="none"></path>
 <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"></path>
 <path d="M0 0h24v24H0z" fill="none"></path>
 </svg>
 </span>
 <span class="ipl-rating-star__rating">9.3</span>
 </div>
 <div class="ipl-rating-interactive small ipl-rating-interactive--no-rating">
 <input class="ipl-rating-interactive__state" data-reftag="ur_rt" data-tconst="tt0111161" id="checkbox-tt0111161" type="checkbox"/>
 <label class="ipl-rating-interactive__star-container" for="checkbox-tt0111161">
 <div class="ipl-rating-star ipl-rating-interactive__star">
 <span class="ipl-rating-star__star">
 <svg class="ipl-icon ipl-star-icon" fill="#000000" height="24" viewbox="0 0 24 24" width="24" xmln

In [14]:
rating =[]
for i in m_imbd1:
    rating.append(i.text.replace("\n","").replace("Rate 1Rate2Rate3Rate4Rate5Rate6Rate7Rate8Rate9Rate10Rate0Error: please try again",""))
rating

['9.30.',
 '9.20.',
 '90.',
 '90.',
 '90.',
 '8.90.',
 '8.90.',
 '8.90.',
 '8.80.',
 '8.80.',
 '8.40.',
 '8.80.',
 '8.80.',
 '8.80.',
 '8.70.',
 '8.70.',
 '8.70.',
 '8.70.',
 '8.70.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.60.',
 '8.50.',
 '8.60.',
 '8.60.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.40.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.50.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.40.',
 '8.60.',
 '8.30.',
 '8.30.',
 '8.40.',
 '8.40.',
 '8.30.',
 '8.30.',
 '8.40.',
 '8.40.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.40.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.',
 '8.30.']

In [15]:
imdb = pd.DataFrame({})
imdb['movies_name']=movies_name[0:100]
imdb['release_year']=year[0:100]
imdb['rating']=rating[0:100]
imdb

movies_name release_year rating
0              The Shawshank Redemption           1.  9.30.
1                         The Godfather       (1994)  9.20.
2                The Godfather: Part II           2.    90.
3                       The Dark Knight       (1972)    90.
4                          12 Angry Men           3.    90.
..                                  ...          ...    ...
95                   North by Northwest       (2019)  8.30.
96                   A Clockwork Orange          49.  8.30.
97                               Snatch       (1942)  8.30.
98  Le fabuleux destin d'Amélie Poulain          50.  8.30.
99                              The Kid       (1988)  8.30.

[100 rows x 3 columns]

In [16]:
len(movies_name),len(release_year ),len(rating)

(100, 100, 100)

Q3.Write a python program to scrap book name, author name, genre and book review of any 5 books from
‘www.bookpage.com’

In [17]:
page= requests.get("https://bookpage.com/reviews")

In [18]:
soup = BeautifulSoup(page.content,"html.parser")

In [19]:
book =soup.find_all("h4",class_="italic")

In [20]:
book_name =[]
for i in  book:
    book_name.append(i.text.replace("\n","").replace("★",""))         
book_name

['  How to Find Your Way in the Dark',
 'Hold Fast Through the Fire',
 'A Good Day for Chardonnay',
 'All’s Well',
 '  A Farewell to Gabo and Mercedes',
 '  Summer Fun',
 '  Sugar Town Queens',
 'We Are Each Other’s Harvest',
 '  Dead Wednesday',
 'The Wreckage of My Presence']

In [21]:
book1 =soup.find_all("p",class_="sans bold")

In [22]:
author_name =[]
for i in  book1:
    author_name.append(i.text.replace("\n",""))         
author_name

['Derek B. Miller',
 'K.B. Wagers',
 'Darynda Jones',
 'Mona Awad',
 'Rodrigo García',
 'Jeanne Thornton',
 'Malla Nunn',
 'Natalie Baszile, Tina Lifford',
 'Jerry Spinelli',
 'Casey Wilson']

In [23]:
book3 =soup.find_all("h4",class_="italic")

In [24]:
genre=[]
for i in book3:
    for j in i.find_all("a",href=True):
        genre.append(j['href'])
genre 

['/reviews/26469-derek-b-miller-how-to-find-your-way-dark-fiction',
 '/reviews/26544-k-b-wagers-hold-fast-through-fire-science-fiction-fantasy',
 '/reviews/26542-darynda-jones-good-day-chardonnay-mystery-suspense',
 '/reviews/26466-mona-awad-alls-well-fiction',
 '/reviews/26490-rodrigo-garcia-farewell-to-gabo-mercedes-nonfiction',
 '/reviews/26474-jeanne-thornton-summer-fun-fiction',
 '/reviews/26536-malla-nunn-sugar-town-queens-ya',
 '/reviews/26462-natalie-baszile-we-are-each-others-harvest-audio',
 '/reviews/26485-jerry-spinelli-dead-wednesday-childrens',
 '/reviews/26461-casey-wilson-wreckage-my-presence-audio']

In [25]:
reivew=[]
for i in genre:
    book =requests.get("https://bookpage.com"+i)
    soup =BeautifulSoup(book.content,'html.parser')
    rev =soup.find("div",class_="article-body")
    reivew.append(rev.text)
reivew  

['\n\nPrepare for surprises galore in How to Find Your Way in the Dark, a rollicking novel that begins with a lonely truck ride in New England in 1938 and follows its characters through a decade of fascinating history. Just when you think the story is heading one way, it veers in another, completely unexpected direction.\nTwelve-year-old Sheldon Horowitz and his father are driving home from Hartford, Connecticut, to Whately, Massachusetts, after honoring the one-year anniversary of Sheldon’s mother’s death. She and her sister died in a horrific movie theater fire in Hartford. And as if that isn’t enough tragedy for the novel’s first 13 pages, a truck purposely forces Sheldon and his father’s car off the road during their return trip, and Sheldon’s father dies.\nReaders of Derek B. Miller’s award-winning thriller, Norwegian by Night, will recognize Sheldon as that novel’s 82-year-old protagonist. As a Tom Sawyer-like boy in How to Find Your Way in the Dark, Sheldon is determined to make

In [26]:
book = pd.DataFrame({})
book['book_name']=book_name
book['author_name']=author_name
book['reivew ']=reivew 
book['genre']=genre[0:10] 

In [27]:
book

book_name                    author_name  \
0    How to Find Your Way in the Dark                Derek B. Miller   
1          Hold Fast Through the Fire                    K.B. Wagers   
2           A Good Day for Chardonnay                  Darynda Jones   
3                          All’s Well                      Mona Awad   
4     A Farewell to Gabo and Mercedes                 Rodrigo García   
5                          Summer Fun                Jeanne Thornton   
6                   Sugar Town Queens                     Malla Nunn   
7         We Are Each Other’s Harvest  Natalie Baszile, Tina Lifford   
8                      Dead Wednesday                 Jerry Spinelli   
9         The Wreckage of My Presence                   Casey Wilson   

                                             reivew   \
0  \n\nPrepare for surprises galore in How to Fin...   
1  \n\nLooking for a quick bit of adrenaline, a s...   
2  \n\nReaders who enjoy murder mysteries with lo...   
3  \n\nMiranda Fitch, the protagonist of Mona Awa...   
4  \n\nRodrigo García is a film and television di...   
5  \n\nIt’s been nine years since the publication...   
6  \n\nExplosive, long-buried family secrets lie ...   
7  \n\nThe audiobook of Natalie Baszile’s We Are ...   
8  \n\nNewbery Medalist Jerry Spinelli has create...   
9  \n\nYou may know Casey Wilson from her brief s...   

                                               genre  
0  /reviews/26469-derek-b-miller-how-to-find-your...  
1  /reviews/26544-k-b-wagers-hold-fast-through-fi...  
2  /reviews/26542-darynda-jones-good-day-chardonn...  
3         /reviews/26466-mona-awad-alls-well-fiction  
4  /reviews/26490-rodrigo-garcia-farewell-to-gabo...  
5  /reviews/26474-jeanne-thornton-summer-fun-fiction  
6     /reviews/26536-malla-nunn-sugar-town-queens-ya  
7  /reviews/26462-natalie-baszile-we-are-each-oth...  
8  /reviews/26485-jerry-spinelli-dead-wednesday-c...  
9  /reviews/26461-casey-wilson-wreckage-my-presen...

Q4.Write a python program to scrape cricket rankings from ‘www.icc-cricket.com’. You have toscrape:

1.Top 10 ODI teams in men’s cricket along with the records for matches, points andrating.

In [28]:
crecket = requests.get("https://www.icc-cricket.com/rankings/mens/team-rankings/odi")
crecket

<Response [200]>

In [29]:
soup = BeautifulSoup(crecket.text, 'html.parser')

In [30]:
crecket1 =soup.find_all("td",class_="table-body__cell rankings-table__team")

In [31]:
Team = []                          
for i in crecket1:
    Team.append(i.text.replace("\n",""))
Team[0:10]

['EnglandENG',
 'AustraliaAUS',
 'IndiaIND',
 'South AfricaSA',
 'PakistanPAK',
 'BangladeshBAN',
 'West IndiesWI',
 'Sri LankaSL',
 'AfghanistanAFG',
 'NetherlandsNED']

In [32]:
crecket2 =soup.find_all("td",class_="table-body__cell u-center-text")

In [33]:
matches1 = []                          
for i in crecket2:
    matches1.append(i.text.replace("\n",""))
matches1

['32',
 '3,793',
 '27',
 '3,109',
 '32',
 '3,624',
 '22',
 '2,267',
 '27',
 '2,524',
 '29',
 '2,639',
 '29',
 '2,458',
 '29',
 '2,303',
 '17',
 '1,054',
 '7',
 '336',
 '23',
 '1,067',
 '17',
 '646',
 '7',
 '258',
 '7',
 '240',
 '5',
 '119',
 '9',
 '190',
 '6',
 '97',
 '8',
 '93',
 '5',
 '0']

In [34]:
match = []                                
for i in range(0,len(matches1),2):
    match.append(matches1[i])
match

['32',
 '27',
 '32',
 '22',
 '27',
 '29',
 '29',
 '29',
 '17',
 '7',
 '23',
 '17',
 '7',
 '7',
 '5',
 '9',
 '6',
 '8',
 '5']

In [35]:
points = []                               
for i in range(-1,len(matches1),2):
    points.append(matches1[i])
points

['0',
 '3,793',
 '3,109',
 '3,624',
 '2,267',
 '2,524',
 '2,639',
 '2,458',
 '2,303',
 '1,054',
 '336',
 '1,067',
 '646',
 '258',
 '240',
 '119',
 '190',
 '97',
 '93',
 '0']

In [36]:
crecket3 =soup.find_all("td",class_="table-body__cell u-text-right rating")

In [37]:
rating = []           
for i in crecket3:
    rating.append(i.text.replace("\n",""))
rating

['119',
 '115',
 '113',
 '103',
 '93',
 '91',
 '85',
 '79',
 '62',
 '48',
 '46',
 '38',
 '37',
 '34',
 '24',
 '21',
 '16',
 '12',
 '0']

In [38]:
top_odi = pd.DataFrame({})
top_odi['Team']=Team[0:10]
top_odi['match']=match[0:10]
top_odi['rating']=rating[0:10]
top_odi['points']=points[0:10]
top_odi

Team match rating points
0      EnglandENG    32    119      0
1    AustraliaAUS    27    115  3,793
2        IndiaIND    32    113  3,109
3  South AfricaSA    22    103  3,624
4     PakistanPAK    27     93  2,267
5   BangladeshBAN    29     91  2,524
6   West IndiesWI    29     85  2,639
7     Sri LankaSL    29     79  2,458
8  AfghanistanAFG    17     62  2,303
9  NetherlandsNED     7     48  1,054

Top 10 ODI Batsmen in men along with the records of their team and rating.

In [39]:
batsmen = requests.get("https://www.icc-cricket.com/rankings/mens/player-rankings/test?at=2021-01-01")
batsmen

<Response [200]>

In [40]:
soup = BeautifulSoup(batsmen.text, 'html.parser')

In [41]:
batsmen1 =soup.find_all("td",class_="table-body__cell name")

In [42]:
batsmen_top = []                          
for i in batsmen1:
    batsmen_top.append(i.text.replace("\n",""))
batsmen_top[0:9]

['Steve Smith',
 'Marnus Labuschagne',
 'Virat Kohli',
 'Joe Root',
 'Rohit Sharma',
 'Rishabh Pant',
 'David Warner',
 'Quinton de Kock',
 'Henry Nicholls']

In [43]:
batsmen1 =soup.find_all("span",class_="table-body__logo-text")

In [44]:
batsmen_team = []                          
for i in batsmen1:
    batsmen_team.append(i.text.replace("\n",""))
batsmen_team[0:9]

['AUS', 'AUS', 'IND', 'ENG', 'IND', 'IND', 'AUS', 'SA', 'NZ']

In [45]:
batsmen2 =soup.find_all("td",class_="table-body__cell u-text-right rating")

In [46]:
batsmen_rating = []                          
for i in batsmen2:
    batsmen_rating.append(i.text.replace("\n",""))
batsmen_rating[0:9]

['891', '878', '812', '797', '759', '752', '724', '717', '714']

In [50]:
top_batsmen = pd.DataFrame({})
top_batsmen["batsmen_top"]=batsmen_top[0:10]
top_batsmen["batsmen_team "]=batsmen_team[0:10]
top_batsmen["batsmen_rating"]=batsmen_rating[0:10]
top_batsmen

batsmen_top batsmen_team  batsmen_rating
0          Steve Smith           AUS            891
1   Marnus Labuschagne           AUS            878
2          Virat Kohli           IND            812
3             Joe Root           ENG            797
4         Rohit Sharma           IND            759
5         Rishabh Pant           IND            752
6         David Warner           AUS            724
7      Quinton de Kock            SA            717
8       Henry Nicholls            NZ            714
9  Ravichandran Ashwin           IND            865

Top 10 ODI bowlers along with the records of their team andrating.

In [127]:
crckt = requests.get("https://www.icc-cricket.com/rankings/mens/player-rankings/odi")
crckt

<Response [200]>

In [128]:
soup = BeautifulSoup(crckt.text, 'html.parser')

In [129]:
crckt1 =soup.find_all("td",class_="table-body__cell name")
crckt1

[<td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/164">Virat Kohli</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/107">Rohit Sharma</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/226">Ross Taylor</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/167">Aaron Finch</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/506">Jonny Bairstow</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/170">David Warner</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/2751">Shai Hope</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/24">Francois du Plessis</a>
 </td>,
 <td class="table-body__cell name">
 <a href="/rankings/mens/player-rankings/834">Quinton de Kock</a>
 </td>,
 <td class="table-body__cell na

In [135]:
bowlers_name =[]
for i in crckt1:
    bowlers_name.append(i.text.replace("\n",""))
bowlers_name[9:]

['Mujeeb Ur Rahman',
 'Chris Woakes',
 'Mehedi Hasan',
 'Matt Henry',
 'Jasprit Bumrah',
 'Josh Hazlewood',
 'Shakib Al Hasan',
 'Kagiso Rabada',
 'Pat Cummins',
 'Mohammad Nabi',
 'Chris Woakes',
 'Rashid Khan',
 'Mitchell Santner',
 'Ben Stokes',
 'Colin de Grandhomme',
 'Imad Wasim',
 'Ravindra Jadeja',
 'Sikandar Raza',
 'Virat Kohli',
 'Rohit Sharma',
 'Ross Taylor',
 'Aaron Finch',
 'Jonny Bairstow',
 'David Warner',
 'Shai Hope',
 'Francois du Plessis',
 'Quinton de Kock',
 'Mujeeb Ur Rahman',
 'Chris Woakes',
 'Mehedi Hasan',
 'Matt Henry',
 'Jasprit Bumrah',
 'Josh Hazlewood',
 'Shakib Al Hasan',
 'Kagiso Rabada',
 'Pat Cummins',
 'Mohammad Nabi',
 'Chris Woakes',
 'Rashid Khan',
 'Mitchell Santner',
 'Ben Stokes',
 'Colin de Grandhomme',
 'Imad Wasim',
 'Ravindra Jadeja',
 'Sikandar Raza']

In [131]:
crckt2 =soup.find_all("span",class_="table-body__logo-text")
crckt2

[<span class="table-body__logo-text">IND</span>,
 <span class="table-body__logo-text">IND</span>,
 <span class="table-body__logo-text">NZ</span>,
 <span class="table-body__logo-text">AUS</span>,
 <span class="table-body__logo-text">ENG</span>,
 <span class="table-body__logo-text">AUS</span>,
 <span class="table-body__logo-text">WI</span>,
 <span class="table-body__logo-text">SA</span>,
 <span class="table-body__logo-text">SA</span>,
 <span class="table-body__logo-text">AFG</span>,
 <span class="table-body__logo-text">ENG</span>,
 <span class="table-body__logo-text">BAN</span>,
 <span class="table-body__logo-text">NZ</span>,
 <span class="table-body__logo-text">IND</span>,
 <span class="table-body__logo-text">AUS</span>,
 <span class="table-body__logo-text">BAN</span>,
 <span class="table-body__logo-text">SA</span>,
 <span class="table-body__logo-text">AUS</span>,
 <span class="table-body__logo-text">AFG</span>,
 <span class="table-body__logo-text">ENG</span>,
 <span class="table-body__

In [136]:
team =[]
for i in crckt2:
    team.append(i.text.replace("\n",""))
team [9:]

['AFG',
 'ENG',
 'BAN',
 'NZ',
 'IND',
 'AUS',
 'BAN',
 'SA',
 'AUS',
 'AFG',
 'ENG',
 'AFG',
 'NZ',
 'ENG',
 'NZ',
 'PAK',
 'IND',
 'ZIM',
 'IND',
 'IND',
 'NZ',
 'AUS',
 'ENG',
 'AUS',
 'WI',
 'SA',
 'SA',
 'AFG',
 'ENG',
 'BAN',
 'NZ',
 'IND',
 'AUS',
 'BAN',
 'SA',
 'AUS',
 'AFG',
 'ENG',
 'AFG',
 'NZ',
 'ENG',
 'NZ',
 'PAK',
 'IND',
 'ZIM']

In [133]:
crckt3 =soup.find_all("td",class_="table-body__cell u-text-right rating")
crckt3

[<td class="table-body__cell u-text-right rating">848</td>,
 <td class="table-body__cell u-text-right rating">817</td>,
 <td class="table-body__cell u-text-right rating">801</td>,
 <td class="table-body__cell u-text-right rating">791</td>,
 <td class="table-body__cell u-text-right rating">775</td>,
 <td class="table-body__cell u-text-right rating">773</td>,
 <td class="table-body__cell u-text-right rating">773</td>,
 <td class="table-body__cell u-text-right rating">766</td>,
 <td class="table-body__cell u-text-right rating">758</td>,
 <td class="table-body__cell u-text-right rating">708</td>,
 <td class="table-body__cell u-text-right rating">700</td>,
 <td class="table-body__cell u-text-right rating">692</td>,
 <td class="table-body__cell u-text-right rating">691</td>,
 <td class="table-body__cell u-text-right rating">683</td>,
 <td class="table-body__cell u-text-right rating">660</td>,
 <td class="table-body__cell u-text-right rating">650</td>,
 <td class="table-body__cell u-text-righ

In [137]:
rating =[]
for i in crckt3:
    rating.append(i.text.replace("\n",""))
rating[9:] 

['708',
 '700',
 '692',
 '691',
 '683',
 '660',
 '650',
 '648',
 '646',
 '294',
 '282',
 '270',
 '268',
 '266',
 '257',
 '256',
 '241',
 '237',
 '848',
 '817',
 '801',
 '791',
 '775',
 '773',
 '773',
 '766',
 '758',
 '708',
 '700',
 '692',
 '691',
 '683',
 '660',
 '650',
 '648',
 '646',
 '294',
 '282',
 '270',
 '268',
 '266',
 '257',
 '256',
 '241',
 '237']

In [140]:
bowler_odi = pd.DataFrame({})
bowler_odi['bowlers_name']=bowlers_name[9:19]
bowler_odi['team']=team[9:19]
bowler_odi['rating']=rating[9:19]
bowler_odi

bowlers_name team rating
0  Mujeeb Ur Rahman  AFG    708
1      Chris Woakes  ENG    700
2      Mehedi Hasan  BAN    692
3        Matt Henry   NZ    691
4    Jasprit Bumrah  IND    683
5    Josh Hazlewood  AUS    660
6   Shakib Al Hasan  BAN    650
7     Kagiso Rabada   SA    648
8       Pat Cummins  AUS    646
9     Mohammad Nabi  AFG    294

Q5.Write a python program to scrape cricket rankings from ‘www.icc-cricket.com’. You have toscrape:

Top 10 ODI teams in women’s cricket along with the records for matches, points andrating.

In [51]:
wom_crecket = requests.get("https://www.icc-cricket.com/rankings/womens/team-rankings/odi")
wom_crecket

<Response [200]>

In [52]:
soup = BeautifulSoup(wom_crecket.text, 'html.parser')

In [53]:
crecket1 =soup.find_all("span",class_="u-hide-phablet")

In [54]:
Team_name = []
for i in crecket1:
    Team_name.append(i.text.replace("\n",""))
Team_name

['Australia',
 'England',
 'South Africa',
 'India',
 'New Zealand',
 'West Indies',
 'Pakistan',
 'Bangladesh',
 'Sri Lanka',
 'Ireland',
 '',
 '',
 '',
 '',
 '']

In [55]:
crecket2 =soup.find_all("td",class_="table-body__cell u-center-text")

In [56]:
matches_info = []
for i in crecket2:
    matches_info.append(i.text.replace("\n",""))
matches_info

['20',
 '2,370',
 '24',
 '2,828',
 '23',
 '2,535',
 '21',
 '1,947',
 '17',
 '1,427',
 '20',
 '1,496',
 '5',
 '306',
 '11',
 '519',
 '2',
 '25']

In [57]:
matches=[]
for i in range(0,len(matches_info),2):
    matches.append(matches_info[i])
matches    

['20', '24', '23', '21', '17', '20', '5', '11', '2']

In [58]:
points=[]
for i in range(1,len(matches_info),2):
    points.append(matches_info[i])
points 

['2,370', '2,828', '2,535', '1,947', '1,427', '1,496', '306', '519', '25']

In [59]:
crecket3 =soup.find_all("td",class_="table-body__cell u-text-right rating")

In [60]:
ratings = []
for i in crecket3:
    ratings.append(i.text.replace("\n",""))
ratings   

['119', '118', '110', '93', '84', '75', '61', '47', '13']

In [63]:
womens_odi = pd.DataFrame({})
womens_odi["Team_name"]=Team_name[0:9]
womens_odi["Team_name "]=Team_name[0:9]
womens_odi["points"]=points[0:9]
womens_odi["matches"]=matches[0:9]
womens_odi

Team_name    Team_name  points matches
0     Australia     Australia  2,370      20
1       England       England  2,828      24
2  South Africa  South Africa  2,535      23
3         India         India  1,947      21
4   New Zealand   New Zealand  1,427      17
5   West Indies   West Indies  1,496      20
6      Pakistan      Pakistan    306       5
7    Bangladesh    Bangladesh    519      11
8     Sri Lanka     Sri Lanka     25       2

Top 10 women’s ODI players along with the records of their team and rating.

In [64]:
wom_top10 = requests.get("https://www.icc-cricket.com/rankings/womens/player-rankings/odi")
wom_top10

<Response [200]>

In [65]:
soup = BeautifulSoup(wom_top10.text, 'html.parser')

In [67]:
crecket1 =soup.find_all("td",class_="table-body__cell name")

In [68]:
player_name = []
for i in crecket1:
    player_name.append(i.text.replace("\n",""))
player_name   

['Lizelle Lee',
 'Alyssa Healy',
 'Tammy Beaumont',
 'Stafanie Taylor',
 'Meg Lanning',
 'Amy Satterthwaite',
 'Natalie Sciver',
 'Smriti Mandhana',
 'Laura Wolvaardt',
 'Megan Schutt',
 'Marizanne Kapp',
 'Shabnim Ismail',
 'Jhulan Goswami',
 'Sophie Ecclestone',
 'Katherine Brunt',
 'Ayabonga Khaka',
 'Poonam Yadav',
 'Ellyse Perry',
 'Ellyse Perry',
 'Stafanie Taylor',
 'Natalie Sciver',
 'Deepti Sharma',
 'Jess Jonassen',
 'Ashleigh Gardner',
 'Dane van Niekerk',
 'Sophie Devine',
 'Katherine Brunt']

In [74]:
crecket2 =soup.find_all("td",class_="table-body__cell nationality-logo")

In [75]:
team_name = []
for i in crecket2:
    team_name.append(i.text.replace("\n",""))
team_name   

['SA',
 'AUS',
 'ENG',
 'WI',
 'AUS',
 'NZ',
 'ENG',
 'IND',
 'SA',
 'AUS',
 'SA',
 'SA',
 'IND',
 'ENG',
 'ENG',
 'SA',
 'IND',
 'AUS',
 'AUS',
 'WI',
 'ENG',
 'IND',
 'AUS',
 'AUS',
 'SA',
 'NZ',
 'ENG']

In [76]:
crecket3 =soup.find_all("td",class_="table-body__cell u-text-right rating")

In [77]:
rating = []
for i in crecket3:
    rating.append(i.text.replace("\n",""))
rating 

['758',
 '756',
 '754',
 '736',
 '723',
 '715',
 '706',
 '701',
 '683',
 '762',
 '747',
 '717',
 '694',
 '668',
 '646',
 '638',
 '617',
 '616',
 '418',
 '394',
 '365',
 '331',
 '307',
 '252',
 '243',
 '242',
 '239']

In [84]:
womens_odi10 = pd.DataFrame({})
womens_odi10["player_name"]=player_name[0:9]
womens_odi10["team_name"]=Team_name[0:9]
womens_odi10["rating"]=points[0:9]
womens_odi10

player_name     team_name rating
0        Lizelle Lee     Australia  2,370
1       Alyssa Healy       England  2,828
2     Tammy Beaumont  South Africa  2,535
3    Stafanie Taylor         India  1,947
4        Meg Lanning   New Zealand  1,427
5  Amy Satterthwaite   West Indies  1,496
6     Natalie Sciver      Pakistan    306
7    Smriti Mandhana    Bangladesh    519
8    Laura Wolvaardt     Sri Lanka     25

Top 10 women’s ODI all-rounder along with the records of their team andrating.

In [85]:
wom_allrounder = requests.get("https://www.icc-cricket.com/rankings/womens/player-rankings/odi/all-rounder")
wom_allrounder

<Response [200]>

In [86]:
soup = BeautifulSoup(wom_allrounder.text, 'html.parser')

In [87]:
crecket1 =soup.find_all("td",class_="table-body__cell rankings-table__name name")

In [88]:
player_name = []
for i in crecket1:
    player_name.append(i.text.replace("\n",""))
player_name

['Ellyse Perry',
 'Stafanie Taylor',
 'Natalie Sciver',
 'Deepti Sharma',
 'Jess Jonassen',
 'Ashleigh Gardner',
 'Dane van Niekerk',
 'Sophie Devine',
 'Katherine Brunt',
 'Amelia Kerr',
 'Hayley Matthews',
 'Jhulan Goswami',
 'Shikha Pandey',
 'Heather Knight',
 'Rumana Ahmed',
 'Nida Dar',
 'Chamari Athapaththu',
 'Sune Luus',
 'Harmanpreet Kaur']

In [89]:
crecket2 =soup.find_all("span",class_="table-body__logo-text")

In [90]:
team_name = []
for i in crecket2:
    team_name.append(i.text.replace("\n",""))
team_name

['AUS',
 'WI',
 'ENG',
 'IND',
 'AUS',
 'AUS',
 'SA',
 'NZ',
 'ENG',
 'NZ',
 'WI',
 'IND',
 'IND',
 'ENG',
 'BAN',
 'PAK',
 'SL',
 'SA',
 'IND']

In [91]:
crecket3 =soup.find_all("td",class_="table-body__cell rating")

In [92]:
rating = []
for i in crecket3:
    rating.append(i.text.replace("\n",""))
rating

['418',
 '394',
 '365',
 '331',
 '307',
 '252',
 '243',
 '242',
 '239',
 '236',
 '215',
 '215',
 '195',
 '181',
 '179',
 '176',
 '164',
 '153',
 '149']

In [94]:
wo_allrounder_odi10 = pd.DataFrame({})
wo_allrounder_odi10["player_name"]=player_name[0:10]
wo_allrounder_odi10["team_name"]=Team_name[0:10]
wo_allrounder_odi10["rating"]=rating[0:10]
wo_allrounder_odi10

player_name     team_name rating
0      Ellyse Perry     Australia    418
1   Stafanie Taylor       England    394
2    Natalie Sciver  South Africa    365
3     Deepti Sharma         India    331
4     Jess Jonassen   New Zealand    307
5  Ashleigh Gardner   West Indies    252
6  Dane van Niekerk      Pakistan    243
7     Sophie Devine    Bangladesh    242
8   Katherine Brunt     Sri Lanka    239
9       Amelia Kerr       Ireland    236

Q6.Write a python program to scrape details of all the mobile phones under Rs. 20,000 listed on Amazon.in. The 
scraped data should include Product Name, Price, Image URL and Average Rating.

In [97]:
amazon = requests.get("https://www.amazon.in/s?k=best+storage+phone+under+20000&adgrpid=64821307194&ext_vrnc=hi&gclid=Cj0KCQjwl_SHBhCQARIsAFIFRVW5J0D6cnOb0J6ubD3y09MGSO7QZhuKCLYyCA_cU8FbbaOpXz4bpsgaAo2IEALw_wcB&hvadid=398060046718&hvdev=c&hvlocphy=20464&hvnetw=g&hvqmt=b&hvrand=1702690209848540774&hvtargid=kwd-900739206667&hydadcr=24511_1971371&tag=googinhydr1-21&ref=pd_sl_qgqu97obp_b")
amazon

<Response [503]>

GETTING 503 RESPONSE AS PER DISCUSSION ON DOUBT CLEAR SEASON WE SHOULD NOT SCRAPE DATA IF RESPONSE IS 503 

Q7.Write a python program to extract information about the local weather from the National Weather Service 
website of USA, https://www.weather.gov/ for the city, San Francisco. You need to extract data about 7 day 
extended forecast display for the city. The data should include period, short description, temperature and 
description.

In [141]:
weather= requests.get("https://forecast.weather.gov/MapClick.php?lat=37.777120000000025&lon=-122.41963999999996#.YQATOY4zbIU")
weather

<Response [200]>

In [142]:
soup = BeautifulSoup(weather.text, 'html.parser')

In [143]:
weather1 =soup.find_all("p",class_="period-name")
weather1

[<p class="period-name">Today<br/><br/></p>,
 <p class="period-name">Tonight<br/><br/></p>,
 <p class="period-name">Wednesday<br/><br/></p>,
 <p class="period-name">Wednesday<br/>Night</p>,
 <p class="period-name">Thursday<br/><br/></p>,
 <p class="period-name">Thursday<br/>Night</p>,
 <p class="period-name">Friday<br/><br/></p>,
 <p class="period-name">Friday<br/>Night</p>,
 <p class="period-name">Saturday<br/><br/></p>]

In [144]:
day_name =[]
for i in weather1:
    day_name.append(i.text.replace("\n","").split())
day_name

[['Today'],
 ['Tonight'],
 ['Wednesday'],
 ['WednesdayNight'],
 ['Thursday'],
 ['ThursdayNight'],
 ['Friday'],
 ['FridayNight'],
 ['Saturday']]

In [145]:
weather2 =soup.find_all("p",class_="short-desc")
weather2 

[<p class="short-desc">Slight Chance<br/>T-storms then<br/>Sunny</p>,
 <p class="short-desc">Mostly Clear<br/>and Breezy<br/>then Partly<br/>Cloudy</p>,
 <p class="short-desc">Partly Sunny<br/>then Mostly<br/>Sunny and<br/>Breezy</p>,
 <p class="short-desc">Partly Cloudy<br/>and Breezy<br/>then Mostly<br/>Cloudy</p>,
 <p class="short-desc">Partly Sunny<br/>then Mostly<br/>Sunny and<br/>Breezy</p>,
 <p class="short-desc">Mostly Clear<br/>and Breezy<br/>then Mostly<br/>Cloudy</p>,
 <p class="short-desc">Mostly Sunny</p>,
 <p class="short-desc">Mostly Cloudy</p>,
 <p class="short-desc">Mostly Sunny</p>]

In [146]:
short_description =[]
for i in weather2:
     short_description.append(i.text.replace("\n",""))
short_description

['Slight ChanceT-storms thenSunny',
 'Mostly Clearand Breezythen PartlyCloudy',
 'Partly Sunnythen MostlySunny andBreezy',
 'Partly Cloudyand Breezythen MostlyCloudy',
 'Partly Sunnythen MostlySunny andBreezy',
 'Mostly Clearand Breezythen MostlyCloudy',
 'Mostly Sunny',
 'Mostly Cloudy',
 'Mostly Sunny']

In [147]:
weather3 =soup.find_all("li",class_="forecast-tombstone")
weather3

[<li class="forecast-tombstone">
 <div class="tombstone-container">
 <p class="period-name">Today<br/><br/></p>
 <p><img alt="Today: A slight chance of showers and thunderstorms before 10am, then a slight chance of showers between 10am and noon.  Mostly cloudy, then gradually becoming sunny, with a high near 73. West southwest wind 8 to 13 mph increasing to 15 to 20 mph in the afternoon. Winds could gust as high as 25 mph.  Chance of precipitation is 20%." class="forecast-icon" src="DualImage.php?i=scttsra&amp;j=few&amp;ip=20" title="Today: A slight chance of showers and thunderstorms before 10am, then a slight chance of showers between 10am and noon.  Mostly cloudy, then gradually becoming sunny, with a high near 73. West southwest wind 8 to 13 mph increasing to 15 to 20 mph in the afternoon. Winds could gust as high as 25 mph.  Chance of precipitation is 20%."/></p><p class="short-desc">Slight Chance<br/>T-storms then<br/>Sunny</p><p class="temp temp-high">High: 73 °F</p></div></li>,

In [148]:
temperature =[]
for i in weather3:
     temperature.append(i.text.replace("\n","").split())
temperature

[['TodaySlight', 'ChanceT-storms', 'thenSunnyHigh:', '73', '°F'],
 ['TonightMostly', 'Clearand', 'Breezythen', 'PartlyCloudyLow:', '57', '°F'],
 ['WednesdayPartly', 'Sunnythen', 'MostlySunny', 'andBreezyHigh:', '71', '°F'],
 ['WednesdayNightPartly',
  'Cloudyand',
  'Breezythen',
  'MostlyCloudyLow:',
  '57',
  '°F'],
 ['ThursdayPartly', 'Sunnythen', 'MostlySunny', 'andBreezyHigh:', '70', '°F'],
 ['ThursdayNightMostly',
  'Clearand',
  'Breezythen',
  'MostlyCloudyLow:',
  '57',
  '°F'],
 ['FridayMostly', 'SunnyHigh:', '70', '°F'],
 ['FridayNightMostly', 'CloudyLow:', '56', '°F'],
 ['SaturdayMostly', 'SunnyHigh:', '68', '°F']]

In [149]:
temperature1 =[]
for i in range(0,len(temperature),1):
     temperature1.append(temperature[i][-2::5])
temperature1

[['73'], ['57'], ['71'], ['57'], ['70'], ['57'], ['70'], ['56'], ['68']]

In [150]:
sfo_weather = pd.DataFrame({})
sfo_weather["day_name"]=day_name
sfo_weather["short_description"]=short_description
sfo_weather["temperature1"]=temperature1
sfo_weather

day_name                         short_description temperature1
0           [Today]           Slight ChanceT-storms thenSunny         [73]
1         [Tonight]   Mostly Clearand Breezythen PartlyCloudy         [57]
2       [Wednesday]    Partly Sunnythen MostlySunny andBreezy         [71]
3  [WednesdayNight]  Partly Cloudyand Breezythen MostlyCloudy         [57]
4        [Thursday]    Partly Sunnythen MostlySunny andBreezy         [70]
5   [ThursdayNight]   Mostly Clearand Breezythen MostlyCloudy         [57]
6          [Friday]                              Mostly Sunny         [70]
7     [FridayNight]                             Mostly Cloudy         [56]
8        [Saturday]                              Mostly Sunny         [68]

Q8.Write a python program to scrape fresher job listings from ‘https://internshala.com/’. It should include job title, 
company name, CTC, and apply date.

In [100]:
internshala = requests.get("https://internshala.com/fresher-jobs/")
internshala

<Response [200]>

In [101]:
soup = BeautifulSoup(internshala.text, 'html.parser')

In [102]:
internshala1 =soup.find_all("div",class_="heading_4_5 profile")

In [103]:
job_title = []                          
for i in internshala1:
    job_title.append(i.text.replace("\n",""))
job_title

['Sales Manager/Omni Sport Leader - Retail ',
 'MERN Stack Developer ',
 'Conversion Rate Optimization Analyst ',
 'Client Service Executive ',
 'Sales Development Representative ',
 'Corporate Sales Associate ',
 'International Business Development Executive ',
 'Product Marketer ',
 'Customer Support Associates ',
 'Content & Customer Care Executive ',
 'Django Developer ',
 'Reactjs Developer ',
 'Web Development Trainee ',
 'Business Development Executive ',
 'Cybersecurity Compliance Officer ',
 'MERN Stack Developer ',
 'Senior Copywriter ',
 'Corporate Sales Associate ',
 'Digital Marketing Associate ',
 'Social Media Marketing Executive ',
 'Business Development Executive ',
 'Operations Manager ',
 'Lead Counselor ',
 'Angular.js Developer ',
 'Digital Marketing Specialist ',
 'Franchise Manager - Sales (Assistant Manager ) ',
 'Graphic Designer ',
 'Project Associate ',
 'Business Development Manager ',
 'Associate Recruiter ',
 'Inside Sales Specialist ',
 'QA Tester ',
 'So

In [104]:
internshala2 =soup.find_all("div",class_="heading_6 company_name")

In [105]:
company_name = []                          
for i in internshala2:
    company_name.append(i.text.strip())
company_name

['Decathlon Sport India Private Limited',
 'Snippt',
 'DataVinci Private Limited',
 'Red Tree Design Studio Private Limited',
 'HelloAR',
 'Best Roadways Limited',
 'Advids',
 'Habitate Technologies Private Limited',
 'Recruit CRM',
 'Revo International LLP',
 'Startxlabs Technologies Private Limited',
 'Startxlabs Technologies Private Limited',
 'Softsensor.ai',
 'Verzeo',
 'S3 Infosoft',
 'Binary Numbers',
 'Ranksoldier International Private Limited',
 'MiM-Essay',
 'BubbleNut Wash',
 'Erikka India',
 'Cosmic Micro Systems Private Limited',
 'Kumi Labs',
 'HOSS - House Of Soft Skills',
 'Systaldyn Consultancy Private Limited',
 'Sahadya Consultants',
 'Regrob',
 'Cynfas',
 'StrategyCo.Global',
 'Narsipur Chemicals Private Limited',
 'Talhive',
 'UniHyr',
 'RavGins International Private Limited (Wobb.ai)',
 'Zolve',
 'Apt Medical Systems Private Limited',
 'CloudGarage',
 'PeakMind',
 'Cafe Tatva Ventures LLP',
 'Bambinos.Live',
 'Open Door Education',
 'Lodha & Bhatt']

In [106]:
internshala3 =soup.find_all("div",class_="item_body")

In [107]:
ctc = []                          
for i in internshala3:
    ctc.append(i.text.replace("\n","").replace("\xa0",""))
ctc

['StartsImmediately                        ',
 '  3.3 - 4.5 LPA                                                    ',
 "21 Aug' 21",
 'StartsImmediately                        ',
 '  4 - 5 LPA                                                    ',
 "26 Aug' 21",
 'StartsImmediately                        ',
 '  5 LPA                                                    ',
 "26 Aug' 21",
 'StartsImmediately                        ',
 '  3 LPA                                                    ',
 "26 Aug' 21",
 'StartsImmediately                        ',
 '  3 - 3.2 LPA                                                    ',
 "26 Aug' 21",
 'StartsImmediately                        ',
 '  3 LPA                                                    ',
 "26 Aug' 21",
 'StartsImmediately                        ',
 '  3 - 4 LPA                                                    ',
 "26 Aug' 21",
 'StartsImmediately                        ',
 '  3.2 - 4.2 LPA                                        

In [108]:
CTC = []
for i in range(1,len(ctc),3):
    CTC.append(ctc[i].strip())
CTC    

['3.3 - 4.5 LPA',
 '4 - 5 LPA',
 '5 LPA',
 '3 LPA',
 '3 - 3.2 LPA',
 '3 LPA',
 '3 - 4 LPA',
 '3.2 - 4.2 LPA',
 '5 - 7 LPA',
 '3 - 3.05 LPA',
 '3.5 - 7.5 LPA',
 '3.5 - 7.5 LPA',
 '3 - 5 LPA',
 '3 - 6 LPA',
 '3 - 3.6 LPA',
 '3 - 7 LPA',
 '3 LPA',
 '4.5 - 6 LPA',
 '3.6 - 5.4 LPA',
 '3 - 5 LPA',
 '3 - 4.8 LPA',
 '3 - 8 LPA',
 '3.8 - 4.5 LPA',
 '3 - 5 LPA',
 '3 - 3.5 LPA',
 '3 - 3.1 LPA',
 '3 LPA',
 '7 - 12 LPA',
 '5 - 7 LPA',
 '3 - 5 LPA',
 '5 LPA',
 '3 - 4 LPA',
 '3 - 8 LPA',
 '3 LPA',
 '3 LPA',
 '6 - 8 LPA',
 '3 - 5.5 LPA',
 '3 - 3.1 LPA',
 '6 - 10 LPA',
 '3.6 - 5 LPA']

In [109]:
apply_date = []
for i in range(2,len(ctc),3):
    apply_date.append(ctc[i].strip())
apply_date 

["21 Aug' 21",
 "26 Aug' 21",
 "26 Aug' 21",
 "26 Aug' 21",
 "26 Aug' 21",
 "26 Aug' 21",
 "26 Aug' 21",
 "25 Aug' 21",
 "25 Aug' 21",
 "25 Aug' 21",
 "25 Aug' 21",
 "25 Aug' 21",
 "25 Aug' 21",
 "25 Aug' 21",
 "25 Aug' 21",
 "23 Aug' 21",
 "23 Aug' 21",
 "25 Aug' 21",
 "23 Aug' 21",
 "23 Aug' 21",
 "22 Aug' 21",
 "25 Aug' 21",
 "22 Aug' 21",
 "22 Aug' 21",
 "22 Aug' 21",
 "22 Aug' 21",
 "22 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "21 Aug' 21",
 "19 Aug' 21",
 "19 Aug' 21"]

In [110]:
fresher_job = pd.DataFrame({})
fresher_job["job_title"]=job_title[0:10]
fresher_job["company_name"]=company_name[0:10]
fresher_job["apply_date"]=apply_date[0:10]
fresher_job["CTC "]=CTC [0:10]
fresher_job

job_title  \
0      Sales Manager/Omni Sport Leader - Retail    
1                          MERN Stack Developer    
2          Conversion Rate Optimization Analyst    
3                      Client Service Executive    
4              Sales Development Representative    
5                     Corporate Sales Associate    
6  International Business Development Executive    
7                              Product Marketer    
8                   Customer Support Associates    
9             Content & Customer Care Executive    

                             company_name  apply_date           CTC   
0   Decathlon Sport India Private Limited  21 Aug' 21  3.3 - 4.5 LPA  
1                                  Snippt  26 Aug' 21      4 - 5 LPA  
2               DataVinci Private Limited  26 Aug' 21          5 LPA  
3  Red Tree Design Studio Private Limited  26 Aug' 21          3 LPA  
4                                 HelloAR  26 Aug' 21    3 - 3.2 LPA  
5                   Best Roadways Limited  26 Aug' 21          3 LPA  
6                                  Advids  26 Aug' 21      3 - 4 LPA  
7   Habitate Technologies Private Limited  25 Aug' 21  3.2 - 4.2 LPA  
8                             Recruit CRM  25 Aug' 21      5 - 7 LPA  
9                  Revo International LLP  25 Aug' 21   3 - 3.05 LPA

Q9.Write a python program to scrape house details from https://www.nobroker.in/ for any location. It should 
include house title, location, area, emi and price

In [112]:
home = requests.get("https://www.nobroker.in/property/sale/gurgaon/Sector%2025?type=RK1,BHK1,BHK2,BHK3,BHK4,BHK4PLUS&searchParam=W3sibGF0IjoyOC40ODI2NzE4MDAwMDAwMSwibG9uIjo3Ny4wNzM5OTYyLCJwbGFjZUlkIjoiQ2hJSmJSMUQ5eGtaRFRrUmxNOUpJdnJHc2k4IiwicGxhY2VOYW1lIjoiR3VyZ2Fvbi1JZmZjbyBDaG93ay0gQWZ0ZXIgSWZmY28gQ2hvd2sgRmx5IE92ZXIifV0=&propertyAge=0&radius=2.0")
home

<Response [200]>

In [113]:
soup = BeautifulSoup(home.text, 'html.parser')

In [114]:
home1 =soup.find_all("h2",class_="heading-6 font-semi-bold nb__1AShY")
home1

[<h2 class="heading-6 font-semi-bold nb__1AShY"><span itemprop="name">3 BHK Apartment  For Sale  In Emaar Mgf Palm Hills In Sector 25</span> <svg aria-hidden="true" class="jss1" focusable="false" role="presentation" viewbox="0 0 24 24"><path d="M0 0h24v24H0z" fill="none"></path><path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"></path></svg></h2>,
 <h2 class="heading-6 font-semi-bold nb__1AShY"><span itemprop="name">3 BHK For Sale in Apartment in Sector 25</span> <svg aria-hidden="true" class="jss1" focusable="false" role="presentation" viewbox="0 0 24 24"><path d="M0 0h24v24H0z" fill="none"></path><path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"></path></svg></h2>,
 <h2 class="heading-6 font-semi-bold nb__1AShY"><span itemprop="name">2 BHK Flat  For Sale  In Gurugram</span> <svg aria-hidden="true" class="j

In [115]:
house_title =[]
for i in home1:
    house_title.append(i.text.replace("\n",""))
house_title 

['3 BHK Apartment  For Sale  In Emaar Mgf Palm Hills In Sector 25 ',
 '3 BHK For Sale in Apartment in Sector 25 ',
 '2 BHK Flat  For Sale  In Gurugram ',
 '4 BHK Flat  For Sale  In Apartment In Sector 25 ',
 '3 BHK Flat  For Sale  In Standalone Building  In Sector 25 ',
 '4 BHK Flat  For Sale  In Apartment In Sector 25 ',
 '4+ BHK Apartment  For Sale  In Suncity Essel Towers In Sector-25 ',
 '4+ BHK For Sale in Suncity Essel Towers in Sector-25 ',
 '3 BHK Apartment  For Sale  In Suncity Essel Towers In Sector-25 ',
 '3 BHK Apartment  For Sale  In Suncity Essel Towers In Sector-25 ']

In [116]:
home2 =soup.find_all("div",class_="nb__2CMjv")
home2

[<div class="nb__2CMjv"><a class="nb__17tcx" href="/emaar-mgf-palm-hills-sector-25_gurgaon-prjt-5ba00ebd714b5687a7dc8aec" target="_blank">Emaar Mgf Palm Hills</a>  <!-- -->DLF Phase 2, Heritage City, Sector 25</div>,
 <div class="nb__2CMjv">Mehrauli-Gurgaon Rd near Privat Hospital</div>,
 <div class="nb__2CMjv">Standalone Building, Sector 25 near  Heritage City Market</div>,
 <div class="nb__2CMjv">DLF Phase 2,Near Digamber Jain Mandir DLF Phase - 2</div>,
 <div class="nb__2CMjv">Mehrauli Gurgaon Service Rd near Delta Square</div>,
 <div class="nb__2CMjv">DLF Phase 2,Near Digamber Jain Mandir DLF Phase - 2</div>,
 <div class="nb__2CMjv"><a class="nb__17tcx" href="/suncity-essel-towers-sector-25_gurgaon-prjt-5ba00ebd714b5687a7dc8cd5" target="_blank">Suncity Essel Towers</a>  <!-- -->Sector-25 and Sector 28, MG Road, Gurgaon, Haryana, INDIA.</div>,
 <div class="nb__2CMjv">Sector-25 and Sector 28, MG Road, Gurgaon, Haryana, INDIA.</div>,
 <div class="nb__2CMjv"><a class="nb__17tcx" href

In [117]:
location_area =[]
for i in home2:
    location_area.append(i.text.replace("\n","").replace("sÂ\xa0",""))
location_area 

['Emaar Mgf Palm Hill DLF Phase 2, Heritage City, Sector 25',
 'Mehrauli-Gurgaon Rd near Privat Hospital',
 'Standalone Building, Sector 25 near  Heritage City Market',
 'DLF Phase 2,Near Digamber Jain Mandir DLF Phase - 2',
 'Mehrauli Gurgaon Service Rd near Delta Square',
 'DLF Phase 2,Near Digamber Jain Mandir DLF Phase - 2',
 'Suncity Essel Tower Sector-25 and Sector 28, MG Road, Gurgaon, Haryana, INDIA.',
 'Sector-25 and Sector 28, MG Road, Gurgaon, Haryana, INDIA.',
 'Suncity Essel Tower Sector-25 and Sector 28, MG Road, Gurgaon, Haryana, INDIA.',
 'Suncity Essel Tower Sector-25 and Sector 28, MG Road, Gurgaon, Haryana, INDIA.']

In [118]:
home3 =soup.find_all("div",class_="font-semi-bold heading-6")
home3

[<div class="font-semi-bold heading-6" id="minRent" itemprop="valueReference" itemscope="" itemtype="http://schema.org/PropertyValue"><meta content="Minimum Rent" itemprop="name"/><meta content="INR" itemprop="unitCode"/><div class="nb__3oNyC">1,575 sqft</div></div>,
 <div class="font-semi-bold heading-6" id="roomType"><span>₹</span>54,448/Month</div>,
 <div class="font-semi-bold heading-6"><meta content="Minimum Deposit" itemprop="name"/><span>₹<!-- -->95 Lacs</span></div>,
 <div class="font-semi-bold heading-6" id="minRent" itemprop="valueReference" itemscope="" itemtype="http://schema.org/PropertyValue"><meta content="Minimum Rent" itemprop="name"/><meta content="INR" itemprop="unitCode"/><div class="nb__3oNyC">2,500 sqft</div></div>,
 <div class="font-semi-bold heading-6" id="roomType"><span>₹</span>1.43 Lacs/Month</div>,
 <div class="font-semi-bold heading-6"><meta content="Minimum Deposit" itemprop="name"/><span>₹<!-- -->2.5 Crores</span></div>,
 <div class="font-semi-bol

In [119]:
area1 =[]
for i in home3:
    area1.append(i.text.replace("\n","").replace("â\x82¹",""))
area1 

['1,575 sqft',
 '54,448/Month',
 '95 Lacs',
 '2,500 sqft',
 '1.43 Lacs/Month',
 '2.5 Crores',
 '900 sqft',
 '63,045/Month',
 '1.1 Crores',
 '2,700 sqft',
 '1.72 Lacs/Month',
 '3 Crores',
 '2,250 sqft',
 '1 Lacs/Month',
 '1.75 Crores',
 '2,700 sqft',
 '1.83 Lacs/Month',
 '3.2 Crores',
 '3,000 sqft',
 '1.43 Lacs/Month',
 '2.5 Crores',
 '5,322 sqft',
 '2.67 Lacs/Month',
 '4.65 Crores',
 '1,354 sqft',
 '94,568/Month',
 '1.65 Crores',
 '1,350 sqft',
 '91,703/Month',
 '1.6 Crores']

In [120]:
area =[]
for i in range(0,len(area1),3):
    area.append(area1[i])
area 

['1,575 sqft',
 '2,500 sqft',
 '900 sqft',
 '2,700 sqft',
 '2,250 sqft',
 '2,700 sqft',
 '3,000 sqft',
 '5,322 sqft',
 '1,354 sqft',
 '1,350 sqft']

In [121]:
emi =[]
for i in range(1,len(area1),3):
    emi.append(area1[i])
emi 

['54,448/Month',
 '1.43 Lacs/Month',
 '63,045/Month',
 '1.72 Lacs/Month',
 '1 Lacs/Month',
 '1.83 Lacs/Month',
 '1.43 Lacs/Month',
 '2.67 Lacs/Month',
 '94,568/Month',
 '91,703/Month']

In [122]:
price =[]
for i in range(2,len(area1),3):
    price.append(area1[i])
price 

['95 Lacs',
 '2.5 Crores',
 '1.1 Crores',
 '3 Crores',
 '1.75 Crores',
 '3.2 Crores',
 '2.5 Crores',
 '4.65 Crores',
 '1.65 Crores',
 '1.6 Crores']

In [126]:
house_details = pd.DataFrame({})
house_details["house_title"]=house_title
house_details["location_area"]=location_area
house_details["area"]=area
house_details["price "]=price
house_details["emi "]=emi
house_details

house_title  \
0  3 BHK Apartment  For Sale  In Emaar Mgf Palm H...   
1          3 BHK For Sale in Apartment in Sector 25    
2                 2 BHK Flat  For Sale  In Gurugram    
3   4 BHK Flat  For Sale  In Apartment In Sector 25    
4  3 BHK Flat  For Sale  In Standalone Building  ...   
5   4 BHK Flat  For Sale  In Apartment In Sector 25    
6  4+ BHK Apartment  For Sale  In Suncity Essel T...   
7  4+ BHK For Sale in Suncity Essel Towers in Sec...   
8  3 BHK Apartment  For Sale  In Suncity Essel To...   
9  3 BHK Apartment  For Sale  In Suncity Essel To...   

                                       location_area        area       price   \
0  Emaar Mgf Palm Hill DLF Phase 2, Heritage City...  1,575 sqft      95 Lacs   
1           Mehrauli-Gurgaon Rd near Privat Hospital  2,500 sqft   2.5 Crores   
2  Standalone Building, Sector 25 near  Heritage ...    900 sqft   1.1 Crores   
3  DLF Phase 2,Near Digamber Jain Mandir DLF Phas...  2,700 sqft     3 Crores   
4      Mehrauli Gurgaon Service Rd near Delta Square  2,250 sqft  1.75 Crores   
5  DLF Phase 2,Near Digamber Jain Mandir DLF Phas...  2,700 sqft   3.2 Crores   
6  Suncity Essel Tower Sector-25 and Sector 28, M...  3,000 sqft   2.5 Crores   
7  Sector-25 and Sector 28, MG Road, Gurgaon, Har...  5,322 sqft  4.65 Crores   
8  Suncity Essel Tower Sector-25 and Sector 28, M...  1,354 sqft  1.65 Crores   
9  Suncity Essel Tower Sector-25 and Sector 28, M...  1,350 sqft   1.6 Crores   

              emi   
0     54,448/Month  
1  1.43 Lacs/Month  
2     63,045/Month  
3  1.72 Lacs/Month  
4     1 Lacs/Month  
5  1.83 Lacs/Month  
6  1.43 Lacs/Month  
7  2.67 Lacs/Month  
8     94,568/Month  
9     91,703/Month